autoconf: don't append -dirty to version

Don't append -dirty to autoconf version number if the buildroot git tree
has uncommited changes.

This script is meant for the autoconf developers, but it also activates
if you build autoconf in a subdirectory of a git tree (E.G. like how it's
commonly done in buildroot).

The affect is that autoconf gets built as being version 2.65-dirty, which
breaks programs (like Python) which explicitly checks for autoconf-2.65.

[Gustavo: update for autoconf 2.69]
Signed-off-by: Peter Korsgaard <jacmet@sunsite.dk>
Signed-off-by: Gustavo Zacarias <gustavo@zacarias.com.ar>

Index: workspace/build-aux/git-version-gen
===================================================================
--- workspace.orig/build-aux/git-version-gen	2024-02-13 21:13:48.378388992 +0100
+++ workspace/build-aux/git-version-gen	2024-02-13 21:13:48.374389007 +0100
@@ -210,7 +210,7 @@
       *) # Append the suffix only if there isn't one already.
           case $v in
             *-dirty) ;;
-            *) v="$v-dirty" ;;
+            #*) v="$v-dirty" ;;
           esac ;;
   esac
 fi
